src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java

```diff
@@ -49,28 +55,64 @@ public class SendToServerProcessor {
   private static final Counter messageNullCounter =
       Metrics.globalRegistry.counter(PROCESSING_ERRORS_COUNTER_NAME, "kind", "message-is-null");
 
+  private static final Counter messageEmptyCounter =
+      Metrics.globalRegistry.counter(
+          PROCESSING_ERRORS_COUNTER_NAME, "kind", "message-batch-is-empty");
+
   private static final Counter filterMatchedCounter =
       Metrics.globalRegistry.counter("fhirtoserver.fhir.filter.matches.total");
 
   private static final Timer filterDurationTimer =
       Metrics.globalRegistry.timer("fhirtoserver.fhir.filter.duration");
 
+  private static final Timer sendingDurationTimer =
+      Metrics.globalRegistry.timer("fhirtoserver.fhir.client.transaction.duration");
+
+  private static final Timer sendingDurationNormalizedTimer =
+      Metrics.globalRegistry.timer(
+          "fhirtoserver.fhir.client.transaction.duration.normalized.by.bundle.size");
+
+  private static final Timer bundleMergingDurationTimer =
+      Metrics.globalRegistry.timer("fhirtoserver.fhir.batch.bundle.merge.duration");
+
+  private static final DistributionSummary sendBundleSizeDistribution =
+      Metrics.globalRegistry.summary("fhirtoserver.fhir.batch.bundle.size");
+
   private final IGenericClient client;
   private final RetryTemplate retryTemplate;
   private final String fhirPathFilterExpression;
   private final Bundle.BundleType overrideBundleType;
   private final FhirPathResourceFilter resourceFilter;
 
-  @Autowired
+  private final boolean isMergeMessageBatchesIntoSingleBundleEnabled;
+  private final String mergeBatchesBundleEntryUniquenessExpression;
+
+  private final FhirBundleMerger fhirBundleMerger;
+
+  private final Optional<Integer> maxMergedBundleSize;
+
   public SendToServerProcessor(
```

Comment:
Ich fand es bei einer bestimmten Anzahl an Values/Env-Vars immer schöner, eine dedizierte Config-Klasse zu erstellen und die Werte auf ein Config-Objekte direkt zu mappen. Würde den Constructor vielleicht etwas schöner machen - aber das ist vielleicht Geschmackssache.

Reply:
geb dir recht. hab 3 parameter in FhirBundleMergerConfig  verschoben